CONTRIBUTING.md

```diff
@@ -70,6 +75,21 @@ For first-time contributors, we suggest looking for issues categorized as [trivi
 
 If you want a hassle-free contribution look for issues categorized as [preapproved](https://issues.dlang.org/buglist.cgi?component=dmd&keywords=preapproved&product=D).
 
+## Refactoring
+
+The purpose of refactoring is to make the code easier to understand. It is not to change the behavior of the
+program, add enhancements, or fix bugs.
+
+- Refactorings must come with a rationale as to why it makes the code better.
+
+- Break up larger refactorings into a series of self-contained PRs rather than one gigantic one.
+
+- Significant refactorings that include bug fixes and/or enhancements will be rejected with a
+request to submit bug fixes and enhancements as separate PRs. Putting them as separate commits
+is not sufficient. Refactorings will be reviewed on their own merits, not conflated with
+other changes.
```

Comment:
```suggestion
- Refactorings must come with a rationale as to why it makes the code better.

- Large refactorings should be done in incremental steps that are easy to review.

- When a refactoring is broken down into multiple PRs, it is acceptable to provide a rationale in the first PR and link to it in subsequent PR.

- Pull requests that do non-trivial refactorings should not include other changes, such as new feature or bug fix.
   While submitting multiple dependent pull requests can be done to help the reviewers judge of the need for a refactoring, any refactoring will be assessed on its own merit.
```

Rationale for the edit:
1) "gigantic" is not very professional;
2) I think we ought to distinguish between "trivial" refactoring (e.g. moving a variable declaration inside a `if` in a single place while fixing the code) and non trivial ones. Trivial refactoring as part of bug fixes / enhancement is common, and forcing people to separate every single line that doesn't directly fix the bug will be a strain on reviewers, contributors and the CI.